packages/react-dom/src/client/ReactDOMComponent.js

```diff
@@ -988,7 +940,7 @@ export function diffHydratedProperties(
       // Validate that the properties correspond to their expected values.
       let serverValue;
       const propertyInfo = getPropertyInfo(propKey);
-      if (suppressHydrationWarning) {
+      if (rawProps[SUPPRESS_HYDRATION_WARNING] === true) {
         // Don't bother comparing. We're ignoring all these warnings.
```

Comment:
We still need to check `rawProps[SUPPRESS_HYDRATION_WARNING] === true` here as per the comment, to skip the heavy comparison code below if the warning was suppressed.